source/common/router/config_impl.h

```diff
@@ -260,7 +260,7 @@ class RetryPolicyImpl : public RetryPolicy {
 
 private:
   std::chrono::milliseconds per_try_timeout_{0};
-  uint32_t num_retries_{};
+  uint32_t num_retries_{1};
```

Comment:
This is more clear I agree, but is this explicitly necessary just for my understanding? We default to 1 when we try to read from proto?

Reply:
It's because here https://github.com/envoyproxy/envoy/blob/master/source/common/router/config_impl.cc#L639-L655 we create a default `RetryPolicyImpl` without parsing any protos when there is no route or vhost level retry policy configured.

Reply:
OK got it. Do you mind adding some comments? Probably worth it given how many questions I had about this. :)

/wait